### PR TITLE
CUCC, CUFLAGS

### PR DESCRIPTION
**Pull request by liruipeng**
1. This PR fixes `CUCC` and `CUFLAGS`, in the cases they are provided by users. Also, make these two variables **precious** (see https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Setting-Output-Variables.html) 

```
Some influential environment variables:
  CUDA_HOME   CUDA home directory
  HYPRE_CUDA_SM
              CUDA architecture
  CUCC        CUDA compiler command
  CUFLAGS     CUDA compiler flags
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  CXX         C++ compiler command
  CXXFLAGS    C++ compiler flags
  FC          Fortran compiler command
  FCFLAGS     Fortran compiler flags
  CPP         C preprocessor

```

2. It changes the warning message from `configure` when without UVM to "It only works for structured solvers **and selected unstructured solvers**"

### CODE REVIEW

**Comment by liruipeng** [comment]
@oseikuffuor1 and I talked about this PR and we made some changes together. Do you and @rfalgout have more suggestions on this? Btw, it will be merged into `gpu_default`, so you will see these again in #336. But it's much easier to see the changes here. Thanks!

File: src/config/configure.in [hunk 1]
```diff
@@ -141,6 +142,7 @@ hypre_using_openmp=no
 hypre_using_device_openmp=no
 hypre_using_insure=no
 hypre_using_cuda=no
+dnl hypre_user_chose_cuda=no
```
Comment (rfalgout) [inline src/config/configure.in]: Just go ahead and remove this.

File: src/config/configure.in [hunk 2]
```diff
@@ -169,7 +171,7 @@ hypre_using_memory_tracker=no
 dnl *********************************************************************
 dnl * Initialize hypre-HIP variables
 dnl *********************************************************************
-hypre_user_chose_hip=no
+dnl hypre_user_chose_hip=no
```
Comment (rfalgout) [inline src/config/configure.in]: Remove

**Review by rfalgout — APPROVED** [review]
This looks good to me.  Thanks!

File: src/config/configure.in (hunk 1 above)
Comment (liruipeng) [reply src/config/configure.in]: Fixed. Thanks!

File: src/config/configure.in (hunk 2 above)
Comment (liruipeng) [reply src/config/configure.in]: Fixed. Thanks!

**Pull request merged**